tools/ackdiscover/service.py

```diff
@@ -58,38 +61,23 @@ def get(writer, repo, model_name):
         api_version=api_version,
     )
 
-    # The api-2.json file can be very large, and we only want to check the
-    # metadata at the top of the file, so here we're going to just read a
-    # portion of the file and manually search for the metadata fields that are
-    # of interest to us.
+    # The api-2.json file can be fairly large (see ec2 api-2.JSON), but it is what it
+    # is. We're going to load it into memory and parse it as JSON.
     writer.debug("[get_service] fetching service information for:", model_name)
     api_model_path = os.path.join(model_path, api_version, "api-2.json")
     with open(api_model_path, "r") as model_file:
-        # Assuming that the information we're looking for is very likely going to
-        # be in the first 100 lines of the file...
-        lines_number = 100
-        head = [next(model_file) for _ in range(lines_number)]
+        model = json.load(model_file)
```

Comment:
ijson parsing might be less greedy of memory. if something breaks, we can look at that.